### DISCUSSION TITLE
Open links in a new window

### DISCUSSION THREAD

**Discussion by viharm** [markedjs/marked#2982] [Q&A] [answered]
Firstly, I love _marked_. As an amateur/hobbyist developer, I can't believe how easy it is to use with minimal configuration. It is also well-documented. Thanks so much.

My question relates to linkification as part of an _[11ty](https://www.11ty.dev)_ project. I simply love how it automatically linkifies URLs. However is it possible configure it to open links in new windows? (`target="_blank"`)?

I am aware that _marked_'s extensibility means this can be done by adding some code. I also believe that _[LinkifyIt](https://github.com/UziTech/marked-linkify-it)_ could possibly used. But I'm trying to find an efficient way without adding too many dependencies.

I tried to search for similar solutions, but couldn't find anything similar. Surely someone must have had this same question?

(PS: I'm aware that _11ty_ uses _mardown-it_ for parsing Markdown, but the latter does not offer in-line parsing (meaning custom filters to remove the `<p>...</p>` tags. And then the complexity of customizing linkified HTML... uggh. Hence I much prefer _marked_)

**Comment by UziTech** [comment, marked as answer]
Thanks for loving marked! 💖

You can override the link renderer to add whatever attributes you want.

```js
const renderer = {
  link(href, title, text) {
    const link = marked.Renderer.prototype.link.call(this, href, title, text);
    return link.replace("<a","<a target='_blank' rel='noreferrer' ");
  }
};

marked.use({
    renderer
});
```

**Reply by romatou** [reply]
Could you also provide renderer for replacing links with React Router `Link` like this: `<Link to="link">title</Link>`?

**Reply by UziTech** [reply]
@romatou you can try [marked-react](https://www.npmjs.com/package/marked-react)

**Reply by romatou** [reply]
It is a good alternative, but it doesn't solve my problem: on link click with `openLinksInNewTab={false}` redirect happens and the state is reloaded. If I go back all data must be fetched again. React Router Link secures that.

**Reply by UziTech** [reply]
Marked renderer returns a string for html. You will have to create an issue on marked-react if you want it to create a react component.

**Reply by romatou** [reply]
Ok, thanks

**Comment by viharm** [comment]
Thanks so much for your prompt response (and apologies for the delay in mine :-P ).

I've resorted to a little bit of a compromise - since I use _marked_ within _11th_ as a "filter", I've used the `replace` method directly to _marked_'s output within the _11ty_ filter.

However I feel that applying this directly to _marked_'s rendered will have significant advantages. However I will try that later. Thanks again.
